### PR TITLE
Annotations and Labels

### PR DESCRIPTION
**Pull request by mathieu-benoit**

### CODE REVIEW

File: resource-definitions/template-driver/annotations/annotations.yaml
Comment (astromechza) [inline resource-definitions/template-driver/annotations/annotations.yaml:36]: I think this line should be de-indented a bit, or moved to just above the `value:` line in the next `add` block

```suggestion
            {{- $port := values .resource.spec.service.ports | first }}

```

File: resource-definitions/template-driver/annotations/annotations.yaml
Comment (mathieu-benoit) [reply resource-definitions/template-driver/annotations/annotations.yaml:21]: Not sure, need to be tested, I remember an issue in the past with `toYaml` are equivalent where boolean in annotations where failing here... I propose we keep this like this as it was already the pattern used here, and we could open a future/new PR for this after this one is merged. WDYT?

Comment (TobiasBabin) [reply resource-definitions/template-driver/annotations/annotations.yaml:21]: I tested your variant @astromechza and it fails when I "quote" the value of any annotation, like this:
```yaml
spec:
  # Set annotations on the Kubernetes Deployment
  deployment:
    annotations:
      deploymentannotationkey: "deploymentannotationvalue"
```

I'm getting `Error: reconciling workload my-workload: extending manifest: unrecognized type: string`
It works fine with the present `range` solution.

I guess we cannot prescribe the way users define the values, and should be robust.

I'm therefore approving now as it works fine. If you see anything we have missed, more than happy to pick it up as a follow-up.

**Review by TobiasBabin — APPROVED** [review]
LGTM - see comment about potential simplification, needs further discussion though.

**Comment by mathieu-benoit** [comment]
Thanks, just merged, and yeah sure, if there is any improvement to apply, let's have another PR for this.

**Pull request merged**
